app/models/concerns/match_decisions/route_thirteen_cancel_reasons.rb

```diff
@@ -0,0 +1,62 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/boston-cas/blob/production/LICENSE.md
+###
+
+module MatchDecisions
+  module RouteThirteenCancelReasons
+    extend ActiveSupport::Concern
+
+    def step_cancel_reasons
+      [].tap do |reasons|
+        reasons << 'Incarcerated'
+        reasons << 'Institutionalized'
+        reasons << 'In Treatment/Recovery Center'
+        reasons << 'Match expired' unless [
+          'MatchDecisions::Thirteen::ThirteenClientMatch',
+          'MatchDecisions::Thirteen::ThirteenMatchAcknowledgement',
+          'MatchDecisions::Thirteen::ThirteenConfirmMatchSuccess',
+        ].include?(type)
```

Comment:
Is this a new pattern?  
I can't decide if I like it.  I was expecting each decision to override/extend the defaults as necessary, but in some ways it's nice to have this collected up.  Maybe a few variables would be helpful?  Something like:
```
pre_hsa_decisions = [
     'MatchDecisions::Thirteen::ThirteenClientMatch',
     'MatchDecisions::Thirteen::ThirteenMatchAcknowledgement',
]
```
That seems to be the most-reused.  There's also `MatchRoute::Base#on_or_after_first_client_step?(decision)` which I think should do roughly the same thing and might be a bit cleaner for most of the repetition.